Source/WTF/wtf/CompactPtr.h

```diff
@@ -33,24 +33,66 @@
 #include <wtf/RawPtrTraits.h>
 #include <wtf/StdLibExtras.h>
 
+#if PLATFORM(COCOA)
+#include <wtf/spi/cocoa/MachVMSPI.h>
+#else
+#define HAVE_36BIT_ADDRESS 0
+#endif
+
 namespace WTF {
 
 template <typename T>
-class CompactPtr {
-    WTF_MAKE_FAST_ALLOCATED;
-public:
-#if HAVE(36BIT_ADDRESS)
+struct BigHeapTypeTraits {
+#if HAVE_36BIT_ADDRESS
     // The CompactPtr algorithm relies on being able to shift
     // a 36-bit address right by 4 in order to fit in 32-bits.
     // The only way this is an ok thing to do without information
     // loss is if the if the address is always 16 bytes aligned i.e.
```

Comment:
This is my typo (embarrassing). It should be `if only if` instead of `is if the if`.